### PR TITLE
[CARBONDATA-708] Fixed Between and Logical AND issue

### PR DESCRIPTION
**Pull request by bhavya411**
- Change CarbonFilters.scala to fix BETWEEN and Logical AND operator
- Add  Unit Test Cases
- Tested on Spark 1.6 and 2.1
- Manual testing done to verify the fix
- Build Success with 1.6 and 2.1

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1327/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala [hunk 1]
```diff
@@ -111,16 +111,24 @@ object CarbonFilters {
...
+      new CarbonLiteralExpression(value, dataType)
     }
 
     createFilter(predicate)
   }
 
 
   // Check out which filters can be pushed down to carbon, remaining can be handled in spark layer.
-  // Mostly dimension filters are only pushed down since it is faster in carbon.
+  // Mostly dimension filters are only pushed down since it is faster in carbo  n.
```
Comment (QiangCai) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala]: redundant blank

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala [hunk 2]
```diff
@@ -111,16 +111,24 @@ object CarbonFilters {
...
     def getCarbonLiteralExpression(name: String, value: Any): CarbonExpression = {
-      new CarbonLiteralExpression(value,
-        CarbonScalaUtil.convertSparkToCarbonDataType(dataTypeOf(name)))
+      val dataTypeOfAttribute = CarbonScalaUtil.convertSparkToCarbonDataType(dataTypeOf(name))
+      val dataType = if (Option(value).isDefined
+                         && dataTypeOfAttribute == DataType.STRING
+                         && value.isInstanceOf[Double]) {
+        DataType.DOUBLE
+      }
+      else {
```
Comment (QiangCai) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala]: take care codestyle

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala [hunk 3]
```diff
@@ -111,16 +111,24 @@ object CarbonFilters {
     }
 
     def getCarbonLiteralExpression(name: String, value: Any): CarbonExpression = {
-      new CarbonLiteralExpression(value,
-        CarbonScalaUtil.convertSparkToCarbonDataType(dataTypeOf(name)))
+      val dataTypeOfAttribute = CarbonScalaUtil.convertSparkToCarbonDataType(dataTypeOf(name))
+      val dataType = if (Option(value).isDefined
+                         && dataTypeOfAttribute == DataType.STRING
+                         && value.isInstanceOf[Double]) {
+        DataType.DOUBLE
```
Comment (QiangCai) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala]: what's the reason to change datatype?

**Comment by ravipesala** [comment]
LGTM, except few style issues raised by david, please fix them

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1349/

**Comment by bhavya411** [comment]
Fixed the coding styles issues as well as the spacing issue

**Pull request closed without merging**